src/pages/logout.js

```diff
@@ -0,0 +1,16 @@
+import { useHistory } from "react-router"
+import { useEffect } from "react/cjs/react.development"
+import { auth } from "../components/constants"
+
+function LogOut(props) {
+  localStorage.removeItem("token")
+  localStorage.removeItem("role")
+  const history = useHistory()
+  useEffect(() => {
+    history.replace(auth)
+    props.setLoggedIn(false)
+  }, [history, props])
+
+  return (<div></div>)
+}
```

Comment:
Move all logic related to logOut into the user service, once logout is done then redirect a user and this all logic should reside in `useEffect` which only once triggers when the component has been mounted.